### PR TITLE
[top] cdc fix and cdc constraints

### PR DESCRIPTION
**Pull request by tjaychen**
**[rv_core_ibex] synchronize timer interrupts**
- timer interrupts do not currently come from the main clock domain
- this PR is a quick fix that synchronizes the interrupts
- long term, we should forward everything through rv_plic only and
  give rv_plic the responsibility of synchronization like it does
  with external interrupts now.

**[top/cdc] Add a few cdc constraints to waive async fifo paths**
Specifically, two types of paths are waived

- fifo rdata outputs that are coupled into integrity errors,
  which then affect whether ibex should cancel or continue
  with the next data fetch

- fifo rdata outputs that are coupled into socket_m1's
  return path selection.  The source ID is used along side
  rvalid to pick a specific dvalid

This fix is hacky right now, as we need to enhance the flow
such that an additional CDC constraint file can be supplied.

### CODE REVIEW

**Review by eunchan — APPROVED** [review]

File: hw/cdc/tools/verixcdc/run-cdc.tcl [hunk 1]
```diff
@@ -101,6 +101,21 @@ set_user_cntl_synchronizer -name opentitan_2ff $prim_2ff_modules
...
+# The path is as follows: asycn_fifo.rdata_o -> data integrity check -> a_valid
+# There are two such paths: One path is a the error pin directly into ibex, and the other is ibex's internal check
+set async_fifo_data [get_pins -of_objects [get_cells -hier * -filter {ref_name == prim_fifo_async}] -filter {name =~ rdata_o*}]
+set_ignore_cdc_paths -name async_fifo_to_ibex_data_err -through_signal $async_fifo_data -through_signal [get_pins top_earlgrey/u_rv_core_ibex/u_core/data_err_i]
+set_ignore_cdc_paths -name async_fifo_to_ibex_ecc_err -through_signal $async_fifo_data -through_signal [get_pins top_earlgrey/u_rv_core_ibex/u_core/u_ibex_core/load_store_unit_i/load_intg_err_o]
+
+# The following paths ignore valid qualification using the response data
+# In the socketm1 module, the returned ID is used to pick which of the original source made the request
+# CDC sees this as rdata directly affecting the control, but the control signal should already dictate whether the rdata is safe to use.
+set_ignore_cdc_paths -name async_fifo_to_ibex_ivalid -through_signal $async_fifo_data -through_signal [get_pins top_earlgrey/u_rv_core_ibex/u_core/instr_rvalid_i]
```
Comment (tjaychen) [inline hw/cdc/tools/verixcdc/run-cdc.tcl:117]: in hindsight, this appears to be too broad. 
By ignoring this path, this now means that any inappropriate linkage between rdata and the rvalid will be ignored. That seems too broad.

Comment (tjaychen) [reply hw/cdc/tools/verixcdc/run-cdc.tcl:117]: @eunchan 
@jeoongp

**Comment by tjaychen** [comment]
@eunchan 
@jeoongp 
are we okay to merge this now? or should we wait until i make this constraint a bit more specific?
let me know what you think.

**Comment by eunchan** [comment]
I am OK with merging this

**Review by jeoongp — APPROVED** [review]
LGTM

**Comment by tjaychen** [comment]
thanks all. I filed a related issue to make sure we come back and clean this up.

**Pull request merged**